src/django_components/tag_formatter.py

```diff
@@ -0,0 +1,269 @@
+                f" Tag must contain only following chars: {VAR_CHARS}"
+            )
+
+    def safe_format_block_start_tag(self, name: str) -> str:
+        """Formats the start tag of a block component. Raises ValueError if the tag is invalid."""
+        tag = self.format_block_start_tag(name)
+        self.validate_tag(tag, "block_start_tag")
+        return tag
+
+    def safe_format_block_end_tag(self, name: str) -> str:
+        """Formats the end tag of a block component. Raises ValueError if the tag is invalid."""
+        tag = self.format_block_end_tag(name)
+        self.validate_tag(tag, "block_end_tag")
+        return tag
+
+    def safe_format_inline_tag(self, name: str) -> str:
+        """Formats the start tag of an inline component. Raises ValueError if the tag is invalid."""
+        tag = self.format_inline_tag(name)
+        self.validate_tag(tag, "inline_tag")
+        return tag
+
+
+# TODO - Update docs that these are names of positional args
+# TODO - Add tests
+class ComponentTagFormatter(TagFormatterABC):
+    """
+    The original django_component's component tag formatter, it uses the `component`
+    tag name, whereas the component name is gives as a string as the second token.
+
+    Example as block:
+    ```django
+    {% component "mycomp" abc=123 %}
+        {% fill "myfill" %}
+            ...
+        {% endfill %}
+    {% endcomponent %}
+    ```
+
+    Example as inlined tag:
+    ```django
+    {% #component "mycomp" abc=123 %}
+    ```
+    """
+
+    def format_block_start_tag(self, name: str) -> str:
+        return "component"
+
+    def format_block_end_tag(self, name: str) -> str:
+        return "endcomponent"
+
+    def format_inline_tag(self, name: str) -> str:
+        return "#component"
+
+    def parse_block_start_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        if tokens[0] != "component":
+            raise TemplateSyntaxError(
+                f"ComponentTagFormatter: Component block start tag parser received tag '{tokens[0]}',"
+                " expected 'component'"
+            )
+        return self._parse_start_or_inline_tag(tokens)
+
+    def parse_inline_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        if tokens[0] != "#component":
+            raise TemplateSyntaxError(
+                f"ComponentTagFormatter: Component block start tag parser received tag '{tokens[0]}',"
+                " expected '#component'"
+            )
+        return self._parse_start_or_inline_tag(tokens)
+
+    def _parse_start_or_inline_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        _, *comp_args = tokens
+
+        if not comp_args:
+            raise TemplateSyntaxError("ComponentTagFormatter: Component tag did not receive tag name")
+
+        # First arg is a kwarg, not a positional arg. Hence look for the "name" kwarg
+        # for component name.
+        if "=" in comp_args[0]:
```

Comment:
This is where the logic from https://github.com/EmilStenstrom/django-components/commit/fbb0f54799eed023dcd7693c967e312db701db67#r145067467 went.